### PR TITLE
[PR]: CDAT Migration: Refactor `aerosol_aeronet` set

### LINKED ISSUE #672: CDAT Migration Phase 2: Refactor `aerosol_aeronet` set

### PR DESCRIPTION
**Pull request by tomvothecoder**
## Description

- Closes #672 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [ ] New and existing unit tests pass with my changes (locally and CI/CD build)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by tomvothecoder — COMMENTED** [review]
Hi @chengzhuzhang, this PR is ready for review. 

This set involved minimal work and will probably have been the easiest to refactor. The `main` branch doesn't save the `.nc` or `.json` files, so we have to do an eye-check for the `.png` plot files (included in PR). As far as I can tell, the plots are identical.

Thanks!

File: e3sm_diags/driver/aerosol_aeronet_driver.py [hunk 1]
```diff
@@ -102,22 +140,24 @@ def interpolate_model_output_to_obs_sites(
...
-        var_intp = np.zeros(nsite)
-        for i in range(nsite):
-            var_intp[i] = f_intp(latloc[i], lonloc[i])
+    # Express lon_loc from 0 to 360.
+    lon_loc[lon_loc < 0.0] = lon_loc[lon_loc < 0.0] + 360.0
+
+    if da_var is not None:
+        lat = xc.get_dim_coords(da_var, axis="Y")
+        lon = xc.get_dim_coords(da_var, axis="X")
+        f_intp = interpolate.RectBivariateSpline(lat.values, lon.values, da_var.values)
```
Comment (tomvothecoder) [inline e3sm_diags/driver/aerosol_aeronet_driver.py:155]: The only actual change in this function. All other changes are variable names.

File: e3sm_diags/driver/aerosol_aeronet_driver.py [hunk 2]
```diff
@@ -25,74 +27,110 @@
...
+            )
+
+            metrics_dict = {
+                "max": da_test.max().item(),
+                "min": da_test.min().item(),
+                "mean": spatial_avg(ds_test, var_key, axis=["X", "Y"]),
+            }
+            aerosol_aeronet_plot.plot(
+                parameter, da_test, test_site_arr, ref_site_arr, metrics_dict
+            )
```
Comment (tomvothecoder) [inline e3sm_diags/driver/aerosol_aeronet_driver.py:95]: For consistency across refactored sets, I moved the metrics dictionary out of the plot function and into the driver.

File: e3sm_diags/driver/aerosol_aeronet_driver.py [hunk 3]
```diff
@@ -25,74 +27,110 @@
...
+                    ds_ref[var_key], var_key
+                )
+            elif run_type == "model_vs_obs":
+                ref_site_arr = interpolate_model_output_to_obs_sites(None, var_key)
+            else:
+                raise ValueError("Invalid run_type={}".format(run_type))
+
+            parameter.output_file = (
+                f"{parameter.ref_name}-{parameter.var_id}-{season}-global"
+            )
```
Comment (tomvothecoder) [inline e3sm_diags/driver/aerosol_aeronet_driver.py:86]: No logic change here except looping over variables then seasons, and using the refactored Dataset class

File: e3sm_diags/plot/cartopy/zonal_mean_2d_stratosphere_plot.py [hunk 4]
```diff
@@ -1,15 +0,0 @@
-import xarray as xr
```
Comment (tomvothecoder) [inline e3sm_diags/plot/cartopy/zonal_mean_2d_stratosphere_plot.py:1]: This file is not needed anymore (`zonal_mean_2d_plot.py` is used directly by this set)

File: e3sm_diags/plot/cartopy/zonal_mean_xy_plot.py [hunk 5]
```diff
@@ -1,138 +0,0 @@
-from __future__ import print_function
```
Comment (tomvothecoder) [inline e3sm_diags/plot/cartopy/zonal_mean_xy_plot.py:1]: This file is note needed anymore. I think this file was accidentally left behind in https://github.com/E3SM-Project/e3sm_diags/pull/752

File: e3sm_diags/driver/zonal_mean_2d_driver.py [hunk 6]
```diff
@@ -18,7 +18,7 @@
     DEFAULT_PLEVS,
     ZonalMean2dParameter,
 )
-from e3sm_diags.plot.cartopy.zonal_mean_2d_plot import plot as plot_func
+from e3sm_diags.plot.zonal_mean_2d_plot import plot as plot_func
```
Comment (tomvothecoder) [inline e3sm_diags/driver/zonal_mean_2d_driver.py:21]: I forgot to move the plotter out of `cartopy` in #774.

**Comment by chengzhuzhang** [comment]
This is nice progress! I will review early next week.

**Review by chengzhuzhang — APPROVED** [review]
Great work. I think this set is ready to ship!

**Pull request merged**